client/src/main/kotlin/io/spine/examples/shareaware/client/MarketPage.kt

```diff
@@ -297,63 +337,478 @@ public class MarketPageModel(client: DesktopClient) {
+                EmptySharesList()
+            } else {
+                SharesList(
+                    listState = listState,
+                    shares = filteredShares,
+                    previousShares = previousShares?.shareList,
+                    model = model
+                )
+            }
+        }
+        VerticalScrollbar(
+            modifier = Modifier
+                .align(Alignment.CenterEnd)
+                .fillMaxHeight(),
+            adapter = rememberScrollbarAdapter(
+                scrollState = listState
+            )
+        )
+    }
+}
+
+/**
+ * Represents the search field.
+ *
+ * @param value the text to be shown in the search field
+ * @param onChange the callback that is triggered when the search value change
+ */
+@Composable
+private fun SearchField(
+    value: String,
+    onChange: (String) -> Unit
+) {
+    Box(
+        modifier = Modifier
+            .padding(top = 10.dp, bottom = 10.dp)
+    ) {
+        Input(
+            value = value,
+            onChange = onChange,
+            placeholder = "Search",
+            isError = false,
+            containerColor = MaterialTheme.colorScheme.secondary,
+            leadingIcon = {
+                Icon(
+                    imageVector = Icons.Default.Search,
+                    contentDescription = "Search",
+                    tint = MaterialTheme.colorScheme.onSecondary,
                     modifier = Modifier
-                        .height(100.dp),
-                    headlineText = {
-                        MainItemContent(share)
-                    },
-                    leadingContent = {
-                        ShareIcon(share)
-                    },
-                    trailingContent = {
-                        ButtonSection(model, share)
-                    },
-                    colors = ListItemDefaults.colors(
-                        containerColor = MaterialTheme.colorScheme.tertiary
-                    )
+                        .size(22.dp)
+                        .padding(end = 5.dp)
                 )
-                Divider(
-                    thickness = 2.dp
+            }
+        )
+    }
+}
+
+/**
+ * Represents the list of shares.
+ *
+ * @param listState the state object to be used to control or observe the list's state
+ * @param shares the list of the actual market shares
+ * @param previousShares the list of the previous market shares
+ * @param model the model of the market page
+ */
+@Composable
+private fun SharesList(
+    listState: LazyListState,
+    shares: List<Share>,
+    previousShares: List<Share>?,
+    model: MarketPageModel
+) {
+    LazyColumn(
+        modifier = Modifier.fillMaxSize(),
+        state = listState
+    ) {
+        shares.forEachIndexed { index, share ->
+            item {
+                ShareItem(
+                    model = model,
+                    share = share,
+                    previousPrice = previousShares?.get(index)?.price
```

Comment:
I think, this access to the `previousShares` by index produced a bug on .gif from the PR description, when price differences are not the same in `ShareList` and on `ShareProfile`.
![image](https://github.com/spine-examples/shareaware/assets/106074440/409e21cd-908e-4f24-90f9-3b17bf1f0b75)